client/ayon_substancedesigner/api/project_creation.py

```diff
@@ -251,6 +290,46 @@ def get_template_filename_from_project(resources_dir,
     return None
 
 
+def resolve_template_path(path, project_name, folder_entity, task_entity):
+    """resolve template path for Substance files
+
+    Args:
+        path (_type_): template path to resolve
+        project_name (str): project name
+        folder_entity (dict): folder entity data
+        task_name (str): task name
+
+    Returns:
+        str: resolved path for Substance template file
+    """
+    anatomy = Anatomy(project_name)
+    fill_data = {
+        key: value
+        for key, value in os.environ.items()
+    }
+
+    fill_data["root"] = anatomy.roots
+    fill_data["project"] = {
+        "name": project_name,
+        "code": anatomy.project_code,
+    }
+    fill_data["folder"] = {
+        "name": folder_entity["name"],
+        "folderType": folder_entity["folderType"],
+        "type": folder_entity["folderType"]
+    }
+    fill_data["task"] = {
+        "name": task_entity["name"],
+        "taskType": task_entity["taskType"],
+        "type": task_entity["taskType"]
+    }
```

Comment:
Why not use `get_template_data` from core, instead of maintaining completely unique 'template data' structure in the codebase here?